packages/mc-scripts/src/commands/build-vite.js

```diff
@@ -0,0 +1,134 @@
+/* eslint-disable react-hooks/rules-of-hooks */
+const fs = require('fs-extra');
+const path = require('path');
+const { build } = require('vite');
+const pluginGraphql = require('@rollup/plugin-graphql');
+const pluginReact = require('@vitejs/plugin-react').default;
+const { useDynamicPublicPath } = require('vite-plugin-dynamic-publicpath');
+const {
+  generateTemplate,
+  htmlScripts,
+} = require('@commercetools-frontend/mc-html-template');
+const {
+  packageLocation: applicationStaticAssetsPath,
+} = require('@commercetools-frontend/assets');
+const paths = require('../config/paths');
+
+const DEFAULT_PORT = parseInt(process.env.HTTP_PORT, 10) || 3001;
+
+const pluginCustomApplication = () => {
+  return {
+    name: 'custom-application',
+    /**
+     * @type {import('vite').IndexHtmlTransformHook}
+     */
+    transformIndexHtml(rawHtml, _ctx) {
+      // Ensure to use the `cdnUrl` value when loading the entry point.
+      // NOTE: with Webpack this is done by setting the `__webpack_public_path__`.
+      const html = rawHtml.replace(
+        new RegExp(`<script type="module"(.*) src="(.*)">`, 'g'),
+        `<script type="module"$1 src="__CDN_URL__$2">`
+      );
+
+      return {
+        html,
+        tags: [
+          {
+            tag: 'script',
+            // Inject the functions to dynamically change the public path.
+            // This is also used for the `cdnUrl` and is the equivalent
+            // of Webpack's `__webpack_public_path__`.
+            children: htmlScripts.publicPath,
+            injectTo: 'body',
+          },
+        ],
+      };
```

Comment:
This transformer is important and it allows to prepend the asset URLs with the `cdnUrl` value.